src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -464,6 +464,11 @@ def config_source_control(resource_group_name, name, repo_url, repository_type=N
             sc = SourceControl(location, name='GitHub', token=git_token)
             client.update_source_control('GitHub', sc)
 
+        # service doesn't support update for being a less likely scenario, but no error is emitted.
+        # we work around it by deleting the existing binding.
+        if show_source_control(resource_group_name, name, slot).repo_url:
+            delete_source_control(resource_group_name, name, slot)
```

Comment:
Does this mean they have to supply all of the source control parameters (set-like behavior instead of update)?